generators/csharp/sdk/src/endpoint/http/HttpEndpointGenerator.ts

```diff
@@ -312,4 +309,347 @@ export class HttpEndpointGenerator extends AbstractEndpointGenerator {
             });
         });
     }
+
+    public generateHttpPagerMethod({
+        serviceId,
+        endpoint,
+        rawClientReference,
+        rawClient
+    }: {
+        serviceId: ServiceId;
+        endpoint: HttpEndpoint;
+        rawClientReference: string;
+        rawClient: RawClient;
+    }): csharp.Method {
+        this.assertHasPagination(endpoint);
+        const endpointSignatureInfo = this.getEndpointSignatureInfo({ serviceId, endpoint });
+        const parameters = [...endpointSignatureInfo.baseParameters];
+        const optionsParamName = this.getRequestOptionsParamNameForEndpoint({ endpoint });
+        const requestOptionsParam = this.getRequestOptionsParameter({ endpoint });
+        const requestOptionsType = requestOptionsParam.type;
+        parameters.push(requestOptionsParam);
+        const itemType = this.getPaginationItemType(endpoint);
+        const return_ = this.getPagerReturnType(endpoint);
+        const snippet = this.getHttpPagerMethodSnippet({ endpoint });
+        return csharp.method({
+            name: this.context.getEndpointMethodName(endpoint),
+            access: csharp.Access.Public,
+            isAsync: false,
+            parameters,
+            summary: endpoint.docs,
+            return_,
+            body: csharp.codeblock((writer) => {
+                const requestParam = endpointSignatureInfo.requestParameter;
+                if (!requestParam) {
+                    throw new Error("Request parameter is required for pagination");
+                }
+                const unpagedEndpointMethodName = this.context.getEndpointMethodName(endpoint);
+                const unpagedEndpointResponseType = getEndpointReturnType({ context: this.context, endpoint });
+                if (!unpagedEndpointResponseType) {
+                    throw new Error("No return type found for endpoint");
```

Comment:
If we have to throw an error, let's be more specific. Something along the lines of:
```suggestion
                    throw new Error("Internal error; a response type is required for pagination endpoints");
```